### PR TITLE
Create and use renv_envvar_exists()

### LINKED ISSUE #1329: Define and use `renv_envvar_exists()` helper for checking existence of envvar
Instead of the `!is.na(Sys.getenv("blah", unset = NA))` pattern.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1329

### CODE REVIEW

File: R/envvar.R [hunk 1]
```diff
@@ -34,3 +34,7 @@ renv_envvar_path_add <- function(envvar, value, prepend = TRUE) {
   new
 
 }
+
+renv_envir_exists <- function(key) {
+  is.na(Sys.getenv(key, NA))
```
Comment (kevinushey) [inline R/envvar.R:39]:
```suggestion
  !is.na(Sys.getenv(key, unset = NA))
```

**Review by kevinushey — COMMENTED** [review]
I think your `renv_envir_exists` implementation is backwards?

**Comment by hadley** [comment]
Ooops, I'll add a test too.

**Comment by kevinushey** [comment]
Also feels somehow terrifying that all of the tests passed regardless 😅

File: R/envvar.R (hunk 1 above)
Comment (hadley) [reply R/envvar.R:39]: I also named the function wrong 😭

**Review by kevinushey — APPROVED** [review]
Feel free to merge after renaming + tidying!

**Pull request merged**